pkg/systemconfig/types.go

```diff
@@ -175,21 +175,21 @@ type registryConf struct {
 type Mirror struct {
 	Location       string   `toml:"location"`
 	PullFromMirror PullType `toml:"pull-from-mirror"`
-	// insecure *bool  `toml:"insecure"`
+	Insecure       *bool    `toml:"insecure"`
 }
 
-func mirrorFor(location string, pullType PullType) Mirror {
+func mirrorFor(location string, pullType PullType, insecure *bool) Mirror {
 	return Mirror{
 		Location:       location,
 		PullFromMirror: pullType,
-		// insecure: insecure,
+		Insecure:       insecure,
 	}
 }
 
 func mirrorsFor(locations []string, pullType PullType) []Mirror {
 	var mirrors []Mirror
 	for _, location := range locations {
-		mirrors = append(mirrors, mirrorFor(location, pullType))
+		mirrors = append(mirrors, mirrorFor(location, pullType, nil))
```

Comment:
Is this an acceptable default value for mirror.Insecure?

Reply:
do we ever pass a non nil value to this function?

Reply:
maybe, once this function becomes a method of SystemConfiSyncer (see https://github.com/openshift/multiarch-tuning-operator/pull/177/files#r1705717318)

```suggestion
		var insecure *bool = nil;
		if  s.registriesConfContent.getRegistryConf(mirror.Location); ok && mrc.Insecure != nil {
		 insecure = mrc.Insecure
		}
		mirrors = append(mirrors, mirrorFor(location, pullType, insecure))
```